ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -0,0 +1,53 @@
+use std::marker::PhantomData;
+use std::mem::transmute;
+use std::ptr::null;
+use std::sync::Arc;
+
+pub unsafe trait RefCounted<'a> where Self: Sized {
+    type Strong;
+    type Borrowed;
+    fn into_strong(x: Arc<Self>) -> Self::Strong;
+    fn null_strong() -> Self::Strong;
+    fn from_borrowed(x: Self::Borrowed) -> &'a Arc<Self>;
+    fn into_borrowed(x: &'a Arc<Self>) -> Self::Borrowed;
+    fn from_borrowed_opt(x: Self::Borrowed) -> Option<&'a Arc<Self>> {
+        let arc = Self::from_borrowed(x);
```

Comment:
If ptr is null, then we have invoked UB by getting an &'a Arc<Self> where the Arc is null, haven't we? I'm worried that this function is unsafe.